status/k8s.go

```diff
@@ -320,17 +325,38 @@ func (k *K8sStatusCollector) status(ctx context.Context) *Status {
 		status.CollectionError(err)
 	}
 
+	ch := make(chan ciliumStatus)
+	var podCount int
+
 	err = k.podStatus(ctx, status, defaults.AgentDaemonSetName, "k8s-app=cilium", func(ctx context.Context, status *Status, name string, pod *corev1.Pod) {
 		if pod.Status.Phase == corev1.PodRunning {
-			s, err := k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
-			status.parseStatusResponse(defaults.AgentDaemonSetName, pod.Name, s, err)
-			status.CiliumStatus[pod.Name] = s
+			podCount++
+			go func() {
+				s, err := k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
```

Comment:
To kind of avoid doing a DoS the kube-apiserver this should be behind a pool of go routines instead of spawning N go routines almost concurrently.

Reply:
Note: [github.com/cilium/workerpool](https://github.com/cilium/workerpool) can help here.

Reply:
Ok added a worker pool. I've extended the pr a bit and made the entire status fetching async.